### PR TITLE
Remove Combine Coverage from PR merge gate

### PR DESCRIPTION
**Pull request by sfc-gh-nkrishna**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   This PR removes Codecov/Combine Coverage from merge gate for PRs. Codecov/Combine Coverage will still be present in the daily precommit job.

### CODE REVIEW

**Comment by sfc-gh-azhan** [comment]
Why "Codecov will still be present in the daily precommit job."?

**Comment by sfc-gh-azhan** [comment]
Why we still keep "precommit test / Combine coverage (pull_request) " in merge gate?

**Comment by sfc-gh-nkrishna** [comment]
> Why "Codecov will still be present in the daily precommit job."?

Yes the daily job still have code coverage: https://github.com/snowflakedb/snowpark-python/blob/main/.github/workflows/daily_precommit.yml#L493-L495.

File: .github/workflows/precommit.yml [hunk 1]
```diff
@@ -515,9 +515,6 @@ jobs:
         with:
           name: overall_cov_xml
           path: .tox/coverage.xml
-      - uses: codecov/codecov-action@v1
```
Comment (sfc-gh-lmukhopadhyay) [inline .github/workflows/precommit.yml:518]: Is there a point to keep the `Combine coverages` and  `Publish html coverage` jobs?

Comment (sfc-gh-azhan) [reply .github/workflows/precommit.yml:518]: You can remove the whole `combine-coverage:` job

**Review by sfc-gh-lmukhopadhyay — APPROVED** [review]

**Review by sfc-gh-azhan — APPROVED** [review]

**Review by sfc-gh-helmeleegy — APPROVED** [review]

**Review by sfc-gh-jrose — APPROVED** [review]

**Pull request merged**